### PR TITLE
Add support for PCF1 to directional / spot lights

### LINKED ISSUE #3194: Implement cheaper shadow sampling
In current implementation, PCF3 is the cheapest available shadow sampling. For mobile, especially older devices, even cheaper filtering could be implemented. PCF2, and even possibly single tap sampling (which is getting used in mobile games) to keep the shadow cost down.

### PR DESCRIPTION
**Pull request by ellthompson**
Updates the light component with support for the PCF1 shadow type when either the directional or spot light type has been selected.

image

Also updates the shadow cascades example with this option.

This is the final change necessary to close #3194

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/scene/light.js
Comment (willeastcott) [reply src/scene/light.js:337]: Sorry, the order was bugging me! 😆

**Comment by slimbuck** [comment]
How about point lights - also supported?

**Comment by ellthompson** [comment]
> How about point lights - also supported?

This is lower priority as omni light shadows aren't cheap to render anyway.

File: src/scene/shader-lib/programs/lit-shader.js
Comment (willeastcott) [inline src/scene/shader-lib/programs/lit-shader.js:1243]: Kinda wish this was a switch statement TBH. But that's not your fault of course! 😄

**Review by willeastcott — APPROVED** [review]
LGTM. Get at least 1 more approval before merging though.

**Comment by mvaligursky** [comment]
I wonder why the shadow looks black and white (binary). When looking at clustered-spot-shadows example (disabled normal maps) with PCF1, there is still filtering taking place, using a single bilinear sample. This should take place in your case as well.
Screenshot 2023-05-15 at 15 58 04

File: src/scene/shader-lib/chunks/lit/frag/shadowStandard.js
Comment (mvaligursky) [inline src/scene/shader-lib/chunks/lit/frag/shadowStandard.js:69]: it seems you're forcing pixel snapping / no filtering here, which I don't think you should? This function should look similarly to what PCF1 shadow does for clustered lights, which is a lot cheaper:

```
    float getShadowSpotClusteredPCF1(SHADOWMAP_ACCEPT(shadowMap), vec3 shadowCoord, vec4 shadowParams) {
        return textureShadow(shadowMap, shadowCoord);
    }

```

Comment (ellthompson) [reply src/scene/shader-lib/chunks/lit/frag/shadowStandard.js]: I've updated this to be in line with clustered light shadows.

File: src/scene/shader-lib/programs/lit-shader.js
Comment (ellthompson) [reply src/scene/shader-lib/programs/lit-shader.js]: I've switched this over.

**Review by mvaligursky — APPROVED** [review]
Looking great!

**Pull request merged**
